bin/petemaild

```diff
@@ -1059,100 +330,25 @@ sub requeue ($$) {
     }
 }
 
-# thing_bad WHAT THING DB MIN MAX
-#
-sub thing_bad ($$$$$$) {
-    my Queue $self = shift;
-    my ($what, $thing, $db, $minint, $maxint) = @_;
-    if (!exists($db->{$thing})) {
-        return 0;
-    } else {
-        my $l = $db->{$thing};
-Log::print('noise', "$what $thing marked as bad ", join(', ', map { int(time() - $_) } @$l), " s ago");
-        # Only one observation, so try again as soon as a short interval
-        # has passed.
-        if (1 == @$l) {
-            if (time() > $l->[0] + $minint) {
-                return 0;
-            } else {
-                return 1;
-            }
-        }
-
-        # Otherwise check after half as long as the thing has been bad already,
-        # or after $maxint, whichever is sooner.
-        my $lastcheck = $l->[@$l - 1];
-        my $interval = ($lastcheck - $l->[0]) / 2;
-        $interval = $maxint if ($interval > $maxint);
-        if ($lastcheck + $interval > time()) {
-            return 1;
-        } else {
-            return 0;
-        }
-    }
-}
-
-# host_bad ADDRESS
-# Should we assume that the host at ADDRESS is bad, or should it be tried
-# again?
-sub host_bad ($$) {
-    my Queue $self = shift;
-    my $addr = shift;
-    return $self->thing_bad('host', $addr, $self->{badhosts},
-                        MIN_HOST_BAD_INTERVAL, MAX_HOST_BAD_INTERVAL);
-}
-
-# domain_bad DOMAIN
-# Should we assume that resolution of the DOMAIN is failing, or should it be
-# tried again?
-sub domain_bad ($$) {
-    my Queue $self = shift;
-    my $domain = shift;
-    return $self->thing_bad('domain', $domain, $self->{baddomains},
-                        MIN_DOMAIN_BAD_INTERVAL, MAX_DOMAIN_BAD_INTERVAL);
-}
-
 # pre_select RDS WRS
 #
 sub pre_select ($$$) {
     my Queue $self = shift;
     my IO::Select $rds = shift;
     my IO::Select $wrs = shift;
 
-    # Do pre-select handling for domain resolver and individual SMTP clients.
-    $self->r()->pre_select($rds, $wrs);
-    foreach my $smtp (@{$self->{ss}}) {
-        $smtp->pre_select($rds, $wrs);
-    }
-
     # Hash of IP addresses to which to send messages to messages to send to
```

Comment:
Remove this comment since the code to which it refers has gone?

Reply:
Weird, how did I miss that.